kotest-framework/kotest-framework-engine/src/jvmTest/kotlin/com/sksamuel/kotest/engine/extensions/spec/AfterEachOrderSpecificationTest.kt

```diff
@@ -0,0 +1,66 @@
+package com.sksamuel.kotest.engine.extensions.spec
+
+import io.kotest.core.extensions.Extension
+import io.kotest.core.listeners.AfterEachListener
+import io.kotest.core.spec.style.FunSpec
+import io.kotest.core.test.TestCase
+import io.kotest.core.test.TestResult
+import io.kotest.matchers.shouldBe
+
+/**
+ * This test seeks to define the order in which [AfterEach] extensions are invoked.
+ */
+class AfterEachOrderSpecificationTest : FunSpec() {
+
+   private var order = ""
+
+   override suspend fun afterEach(testCase: TestCase, result: TestResult) {
+      order += "a"
+   }
+
+   override fun extensions(): List<Extension> {
+      return listOf(
+         object : AfterEachListener {
+            override suspend fun afterEach(testCase: TestCase, result: TestResult) {
+               order += "f"
+            }
+         },
+         object : AfterEachListener {
+            override suspend fun afterEach(testCase: TestCase, result: TestResult) {
+               order += "g"
+            }
+         },
+      )
+   }
+
+   init {
+
+      afterProject {
+         order shouldBe "fgacbfgaedcbfgaedcb"
```

Comment:
will this test become more readable if single characters are replaced with meaningful names, so that `order` is a comma-separated list like this: `listener0, listener1,...`?

Reply:
done, good suggestion, much clearer